src/services/contracts/deployments.ts

```diff
@@ -28,6 +29,31 @@ export const hasCanonicalDeployment = (deployment: SingletonDeploymentV2 | undef
   return networkAddresses.some((networkAddress) => sameAddress(canonicalAddress, networkAddress))
 }
 
+/**
+ * Checks if any of the deployments returned by the `getDeployments` function for the given `network` and `versions` contain a deployment for the `contractAddress`
+ *
+ * @param getDeployments function to get the contract deployments
+ * @param contractAddress address that should be included in the deployments
+ * @param network chainId that is getting checked
+ * @param versions supported Safe versions
+ * @returns true if a matching deployment was found
+ */
+export const hasMatchingDeployment = (
```

Comment:
Should we test this function on its own as well or do you think it has enough coverage through testing `useCompatibleNetworks`?

Reply:
According to the coverage report those lines are covered so I'd say it should be enough.